File: POO_2026_1/src/POO/Pessoa.java
package POO;

import java.util.ArrayList;

public class Pessoa {
  private int idPessoa;
  private String nome;
  private char sexo;
  private Data dtNasc; // data de nascimento (relacionamento de associação)

    public Pessoa(int idPessoa, String nome, char sexo, Data dtNasc) {
        this.idPessoa = idPessoa;
        this.nome = nome;
        this.sexo = sexo;
        this.dtNasc = dtNasc;
    }

    public int getIdPessoa() {
        return idPessoa;
    }

    public void setIdPessoa(int idPessoa) {
        this.idPessoa = idPessoa;
    }


    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public char getSexo() {
        return sexo;
    }

    public void setSexo(char sexo) {
        this.sexo = sexo;
    }

    public Data getDtNasc() {
        return dtNasc;
    }

    public void setDtNasc(Data dtNasc) {
        this.dtNasc = dtNasc;
    }
    
    public static void listarPessoas(ArrayList<Pessoa> pessoas) {
      // for-each
      for(Pessoa pessoa: pessoas) {
        System.out.println(pessoa.toString());
      }
    }

    // Polimorfismo de Sobreposição
    @Override
    public String toString() {
        String result;
        
        result = String.format("id. Pessoa...: %d\n", idPessoa) +
                 String.format("Nome.........: %s\n", nome) +
                 String.format("Sexo.........: %c\n", sexo) +
                 String.format("Nascimento...: %s\n", dtNasc.toString(true));
        
        return result;
    }
    
    
    
    
  
  

}

File: CPP/POO_CPP/CadastrarPessoa_CPP/Pessoa.cpp
#include <iostream>
#include "Pessoa.h"
#include "C:\Meus Documentos\Professor do DAELN\Aulas\TecProg\CPP\POO_CPP\MinhaData_CPP\Data.h"

using namespace std;

Pessoa::Pessoa(int idPessoa, string nome, char sexo, Data *dtNasc) {
  this->idPessoa = idPessoa;
  this->nome = nome;
  this->sexo = sexo;
  this->dtNasc = dtNasc;
}

Pessoa::~Pessoa() {
}

int Pessoa::getIdPessoa() {
  return(idPessoa);
}

void Pessoa::setIdPessoa(int idPessoa) {
  this->idPessoa = idPessoa;
}

string Pessoa::getNome() {
  return nome;
}

void Pessoa::setNome(string nome) {
  this->nome = nome;
}

char Pessoa::getSexo() {
  return sexo;
}

void Pessoa::setSexo(char sexo) {
  this->sexo = sexo;
}

Data *Pessoa::getDtNasc() {
  return dtNasc;
}

void Pessoa::setDtNasc(Data *dtNasc) {
  this->dtNasc = dtNasc;
}

void Pessoa::listarPessoas(vector<Pessoa*> x) {
  for(Pessoa *item: x) {
    cout << item->toString() << endl << endl;
  }
}

string Pessoa::toString() {
  char result[200];
  sprintf(result, "Id. Pessoa...: %d\n"
                  "Nome.........: %s\n"
                  "Sexo.........: %c\n"
                  "Nascimento...: %s",
    idPessoa,
    nome.c_str(),
    sexo,
    dtNasc->toString(true).c_str());

  return(result);
}
